src/core/qgismobileapp.cpp

```diff
@@ -800,13 +800,57 @@ void QgisMobileapp::readProjectFile()
 
   emit loadProjectEnded();
 
+  QgsRectangle projectExtent = getProjectExtent( mProjectFilePath );
+  if ( !projectExtent.isEmpty() )
+    extent = projectExtent;
+
   if ( !extent.isEmpty() && extent.width() != 0.0 )
   {
     // Add a bit of buffer so elements don't touch the map edges
     emit setMapExtent( extent.buffered( extent.width() * 0.02 ) );
   }
 }
 
+QgsRectangle QgisMobileapp::getProjectExtent( const QString &path )
+{
+  QgsRectangle extent;
+
+  QFileInfo fi( path );
+  if ( fi.exists() )
+  {
+    QSettings settings;
+    if ( fi.size() == settings.value( QStringLiteral( "/qgis/projectExtent/%1/filesize" ).arg( path ), 0 ).toLongLong() )
+    {
+      QStringList parts = settings.value( QStringLiteral( "/qgis/projectExtent/%1/extent" ).arg( path ), QString() ).toString().split( '|' );
+      if ( parts.size() == 4 )
+      {
+        extent.setXMinimum( parts[0].toDouble() );
+        extent.setXMaximum( parts[1].toDouble() );
+        extent.setYMinimum( parts[2].toDouble() );
+        extent.setYMaximum( parts[3].toDouble() );
+      }
+    }
+  }
+
+  return extent;
+}
+
+void QgisMobileapp::saveProjectExtent( const QgsRectangle &extent )
```

Comment:
```suggestion
void QgisMobileapp::saveProjectMetainfo( const QgsProjectMetaInfo &info )
```

I wonder if it would make sense to externalize this, so we can add other things to the mix like last used values across sessions etc?

Reply:
Hm, maybe (like last used theme for e.g.). But I'd rather cross that bridge if/when we get there? It'd add quite a lot of code (e.g. coming up with an info class/struct, passing that onto QML as a static object or something, etc. etc.)

Reply:
Ok, let's quickly celebrate that we don't promise a stable API here